transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -268,6 +270,9 @@ private HttpResponse doSend(HttpRequest request) throws IOException {
       // JsonWriter often writes one character at a time.
       dataEncoder.encode(
           request.requestBody, new BufferedWriter(new OutputStreamWriter(outputStream)));
+    } catch (ConnectException | UnknownHostException e) {
```

Comment:
Thinking out loud: This seems like a pretty conservative approach, as opposed to treating all IO Exceptions as retry-able.

Reply:
Yeah exactly - the reason we did this is there are legit IO Exceptions that would not be retryable, eg. encoding errors of some sort